src/cache/lru_evictor.cpp

```diff
@@ -82,14 +88,24 @@ bool LRUEvictor::LRUPartition::do_evict(const uint32_t record_fid,
   auto it = std::begin(m_list);
   while (will_fill(needed_size) && (it != std::end(m_list))) {
     CacheRecord &rec = *it;
-
+    bool eviction_failed{true};
     /* return the next element */
-    if (!rec.is_pinned() && m_evictor->can_evict_cb(record_fid)(rec)) {
-      m_filled_size -= rec.size();
+    if (!rec.is_pinned() && (!m_evictor->can_evict_cb(record_fid) || m_evictor->can_evict_cb(record_fid)(rec))) {
+      auto const rec_size = rec.size();
       it = m_list.erase(it);
-    } else {
-      ++count;
-      it = std::next(it);
+      if (m_evictor->post_eviction_cb(record_fid) && !m_evictor->post_eviction_cb(record_fid)(rec)) {
+          // If the post eviction callback fails, we need to reinsert the record
+          // back into the list.
+          it = m_list.insert(it, rec);
+      } else {
+        eviction_failed = false;
+        m_filled_size -= rec_size;
+      }
+    }
+    
+    if (eviction_failed) {
+        ++count;
```

Comment:
seems this count is never used anywhere? What is the purpose of this increment?

Reply:
This count is used for logging when eviction fails